java/java.project.ui/src/org/netbeans/spi/java/project/support/ui/CreateJavaClassFileFromClipboard.java

```diff
@@ -0,0 +1,258 @@
+                return t;
+            }
+
+            if (classContent.getPackageName() != null) {
+                copiedData = removePackage(copiedData, classContent.getPackageName());
+            }
+            try (BufferedWriter bw = new BufferedWriter(new FileWriter(fileName))) {
+                String packageLocation = getPackageNameFromFile(fileName);
+                if (packageLocation != null && !packageLocation.isEmpty()) {
+                    copiedData = "package " + packageLocation + ";\n" + copiedData;// NOI18N
+                }
+                bw.write(copiedData);
+            }
+
+        } catch (UnsupportedFlavorException ex) {
+            Exceptions.printStackTrace(ex);
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        return t;
+    }
+    
+    
+     private static class DeadlockTask implements Task<CompilationController> {
+
+        JavaSource.Phase phase;
+        CompilationInfo info;
+
+        public DeadlockTask(JavaSource.Phase phase) {
+            assert phase != null;
+            this.phase = phase;
+        }
+
+        public void run(CompilationController info) {
+            try {
+                info.toPhase(this.phase);
+                this.info = info;
+            } catch (IOException ioe) {
+            }
+        }
+
+    }
+
+    private ClassContent extractPackageAndClassName(String copiedData) {
+        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+        StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
+
+        String publicFirstClassName = null;
+        String nonPublicFirstClassName = null;
+        String packageName = null;
+        int counter = 0;
+        JavaCompiler.CompilationTask task = compiler.getTask(null, fileManager, null, null, null, Arrays.asList(new MyFileObject(copiedData)));
+        parse:
+        try {
+            for (CompilationUnitTree compilationUnitTree : ((JavacTask) task).parse()) {
+                packageName = compilationUnitTree.getPackageName() != null
+                        ? compilationUnitTree.getPackageName().toString() : null;
+                for (Tree tree : compilationUnitTree.getTypeDecls()) {
+                    if (tree instanceof ClassTree) {
+                        final ClassTree classTree = (ClassTree) tree;
+                        if (classTree.toString().trim().startsWith(PUBLIC_MODIFIER)) {
+                            publicFirstClassName = classTree.getSimpleName().toString();
+                            break parse;
+                        }
+                        if (counter == 0) {
+                            nonPublicFirstClassName = classTree.getSimpleName().toString();
+                            counter++;
+                        }
+                    }
+                }
+            }
+        } catch (Exception ex) {
```

Comment:
What exceptions are expected here?

Reply:
Expecting IOExection here, changed this to IOException.